dig.go

```diff
@@ -253,13 +253,27 @@ type node struct {
 	provides reflect.Type
 	ctor     interface{}
 	ctype    reflect.Type
-	deps     []reflect.Type
+	deps     []dep
+}
+
+type dep struct {
```

Comment:
you're half way to swapping out `map[reflect.Type]` for `map[key]` to support named types :)